### PR TITLE
[Platform] Add convenience methods to access string name

### PR DESCRIPTION
**Pull request by mrackwitz**
Nice to have for https://github.com/CocoaPods/CocoaPods/pull/4146.
/c @segiddins.

### CODE REVIEW

File: lib/cocoapods-core/platform.rb [hunk 1]
```diff
@@ -200,6 +200,19 @@ def supports_dynamic_frameworks?
...
+    # @return [String] The string that describes the #symbolic_name.
+    #
+    def string_name
+      self.class.string_name(symbolic_name)
+    end
+
+    # @return [String] The string that describes the #symbolic_name,
+    #         which doesn't contain spaces and is so safe to use in
+    #         paths which might not be quoted or escaped consequently.
+    def safe_string_name
```
Comment (segiddins) [inline lib/cocoapods-core/platform.rb]: `trimmed_string_name`, maybe?

File: spec/platform_spec.rb [hunk 2]
```diff
@@ -33,6 +33,20 @@ module Pod
         @platform.name.should == :ios
       end
 
+      it "exposes it's name as string" do
```
Comment (segiddins) [inline spec/platform_spec.rb]: No '

**Comment by segiddins** [comment]
👍🏻 other than comments.

File: lib/cocoapods-core/platform.rb (hunk 1 above)
Comment (mrackwitz) [inline lib/cocoapods-core/platform.rb]: It's not trimmed. It just removes the space in `OS X`.

Comment (segiddins) [inline lib/cocoapods-core/platform.rb]: ok, maybe trimmed was a bad example, but `safe` doesn't semantically convey the transformation either

**Comment by segiddins** [comment]
@mrackwitz ping

File: lib/cocoapods-core/platform.rb (hunk 1 above)
Comment (mrackwitz) [inline lib/cocoapods-core/platform.rb]: I don't have any better idea than that. I think the return type documentation comment gives a good sense of what is meant by it.

File: spec/platform_spec.rb (hunk 2 above)
Comment (mrackwitz) [inline spec/platform_spec.rb]: Ok, I'll fix them altogether.

File: lib/cocoapods-core/platform.rb (hunk 1 above)
Comment (segiddins) [inline lib/cocoapods-core/platform.rb]: ok

**Comment by segiddins** [comment]
:+1:

**Pull request merged**